src/test/java/org/mule/tools/apikit/MainAppScaffolderWithExistingConfigApiSyncTest.java

```diff
@@ -102,17 +107,88 @@ public void reScaffoldDifferntVersions() throws Exception {
+    compareFiles(secondScaffoldingResult, secondScaffoldingGlobalPathV2);
+  }
+
+  private List<MuleConfig> generateSecondScaffoldingConfigurations(String secondScaffoldingApiPathV1,
+                                                                   String secondScaffoldingGlobalPathV1)
+      throws Exception {
+    List<MuleConfig> configurations = new ArrayList<MuleConfig>();
+    String[] existingMuleConfigPaths = {secondScaffoldingGlobalPathV1, secondScaffoldingApiPathV1};
+    for (String s : existingMuleConfigPaths) {
+      try (InputStream muleConfigIS = getResourceAsStream(s)) {
+        MuleConfig existingMuleConfig = MuleConfigBuilder.fromStream(muleConfigIS);
+        existingMuleConfig.setName(extractName(s));
+        configurations.add(existingMuleConfig);
+      }
+    }
+    return configurations;
+  }
+
+  private void firstScaffolding(String raml, String ramlFolderV1, String firstScaffoldingPath, String rootRamlResourceUrl,
+                                List<MuleConfig> muleConfigs)
+      throws IOException, SAXException {
+    ScaffoldingResult result = scaffoldApiSync(raml, ramlFolderV1, rootRamlResourceUrl, muleConfigs);
+    String expected = APIKitTools.readContents(getResourceAsStream(firstScaffoldingPath));
+    String generated = APIKitTools.readContents(result.getGeneratedConfigs().get(0).getContent());
+    Diff diff = XMLUnit.compareXML(expected, generated);
+    assertTrue(diff.identical());
+  }
 
-    String secondScaffoldingExpected = APIKitTools.readContents(getResourceAsStream("rescaffolding-apisync-version/v2/api.xml"));
-    String secondScaffoldingGenerated =
-        APIKitTools.readContents(secondScaffoldingResult.getGeneratedConfigs().get(0).getContent());
-    Diff secondScaffoldingDiff = XMLUnit.compareXML(secondScaffoldingExpected, secondScaffoldingGenerated);
+  private void compareFiles(ScaffoldingResult secondScaffoldingResult, String expectedPath) throws IOException, SAXException {
+    String secondScaffoldingExpectedFile = APIKitTools.readContents(getResourceAsStream(expectedPath));
+    String secondScaffoldingGeneratedFile =
+        createSecondScaffoldingGeneratedFile(secondScaffoldingResult, extractName(expectedPath));
+    Diff secondScaffoldingDiff = XMLUnit.compareXML(secondScaffoldingExpectedFile, secondScaffoldingGeneratedFile);
     assertTrue(secondScaffoldingDiff.identical());
   }
 
+  private String createSecondScaffoldingGeneratedFile(ScaffoldingResult secondScaffoldingResult, String fileName)
+      throws IOException {
+    InputStream api = secondScaffoldingResult.getGeneratedConfigs().stream().filter(config -> config.getName().contains(fileName))
+        .findFirst().get().getContent();
+    return APIKitTools.readContents(api);
+  }
+
+  private String extractName(String path) {
+    String name = "";
+    String[] pathParts = path.split("/");
+    name = pathParts[pathParts.length - 1];
+    return name;
+  }
```

Comment:
lastIndexOf("/")

Reply:
I tried with this and failed, I need the last index from the array instead of the string.